app/migrations/Version20180502142950.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace Migrations;
+
+use AppBundle\Donation\DonationRequestUtils;
+use AppBundle\Entity\Donation;
+use Doctrine\DBAL\Migrations\AbstractMigration;
+use Doctrine\DBAL\Schema\Schema;
+use Ramsey\Uuid\Uuid;
+use Symfony\Component\DependencyInjection\ContainerAwareInterface;
+use Symfony\Component\DependencyInjection\ContainerAwareTrait;
+
+class Version20180502142950 extends AbstractMigration implements ContainerAwareInterface
+{
+    use ContainerAwareTrait;
+
+    public function up(Schema $schema): void
+    {
+        if (!$this->container) {
+            throw new \LogicException('We need the container to get DonationRequestUtils service');
+        }
+        $donationRequestUtils = $this->container->get(DonationRequestUtils::class);
+
+        $this->addSql('CREATE INDEX donation_uuid_idx ON donations (uuid)');
+        $this->addSql('ALTER TABLE donations ADD status VARCHAR(25) DEFAULT NULL, CHANGE created_at created_at DATETIME NOT NULL COMMENT \'(DC2Type:datetime_immutable)\', CHANGE donated_at updated_at DATETIME DEFAULT NULL, ADD paybox_order_ref VARCHAR(255) DEFAULT NULL');
+        $results = $this->connection->executeQuery('SELECT uuid, CONCAT(first_name, \' \', last_name) AS full_name FROM donations')->fetchAll();
+
+        foreach ($results as ['uuid' => $uuid, 'full_name' => $fullName]) {
+            $slug = $donationRequestUtils->buildDonationReference(Uuid::fromString($uuid), $fullName);
+            $this->addSql(
+                'UPDATE donations SET paybox_order_ref = :slug WHERE uuid = :uuid',
+                ['slug' => $slug, 'uuid' => $uuid]
+            );
+        }
+        $this->addSql(
+            'UPDATE donations SET donations.status = :status WHERE donations.finished = 1',
+            ['status' => Donation::STATUS_FINISHED]
+        );
+        $this->addSql(
+            'UPDATE donations SET donations.status = :status WHERE donations.finished = 0',
+            ['status' => Donation::STATUS_WAITING_CONFIRMATION]
+        );
+        $this->addSql(
+            'UPDATE donations SET donations.status = :status WHERE donations.duration != 0',
+            ['status' => Donation::STATUS_SUBSCRIPTION_IN_PROGRESS]
+        );
+        $this->addSql('UPDATE donations SET donations.updated_at = donations.created_at WHERE donations.updated_at IS NULL');
```

Comment:
When the status or any other things change I think it's good to know when ?

Reply:
Sorry it's not my question.

I meant, why do you need to set this field as required while it was not required before. It feels like an unneeded change.

Reply:
It's to have the last update, donated_at transformed to updated_at, donated_at can be null, updated_at will be not null because we want to know when was the last change. At the entity creation updated_at have the same value to created_at

Reply:
Ok fair enough, thank you! 
I missed some stuff :D